### PR TITLE
[seo] optimize titles and descriptions

### LINKED ISSUE #3188: SEO: Optimise titles and descriptions for all pages
### Feature description

We should review the title and descriptions tags for all pages and optimise them.

For most pages, the title is already defined but the description is not.

### Additional context

- Meta descriptions according to Google: https://developers.google.com/search/docs/appearance/snippet#meta-descriptions
- Influencing your title links in search results: https://developers.google.com/search/docs/appearance/title-link

### PR DESCRIPTION
**Pull request by techtolentino**
## Changes

Resolves #3188 

## Implementation details

Using `useSeoMeta` composable for the following pages:

- [x] /events/fall-fest
- [x] /events/summer-school-2022
- [x] /learn
- [x] /learn/course/algorithm-design
- [x] /learn/course/basics-quantum-information
- [x] /learn/course/fundamentals-quantum-algorithms
- [x] /learn/course/games-and-demos
- [x] /learn/course/introduction-course
- [x] /learn/course/legacy-content
- [x] /learn/course/machine-learning-course
- [x] /learn/course/multiple-qubits-and-entanglement
- [x] /learn/course/prerequisites
- [x] /learn/course/quantum-algorithms-for-applications
- [x] /learn/course/quantum-computing-labs
- [x] /learn/course/quantum-hardware-circuits
- [x] /learn/course/quantum-hardware-pulses
- [x] /learn/course/quantum-protocols-and-quantum-algorithms
- [x] /learn/course/quantum-states-and-qubits
- [x] /learn/summer-school/introduction-to-quantum-computing-and-quantum-hardware-2020
- [x] /learn/summer-school/quantum-computing-and-quantum-learning-2021
- [x] /learn/summer-school/quantum-simulation-summer-school-2022

## How to read this PR

- review use of composable
- review copy updates
- feel free to use a SEO Meta Tag extension checker like this [one](https://chrome.google.com/webstore/detail/meta-seo-inspector/ibkclpciafdglkjkcibmohobjkcfkaef) to verify 

### Homepage
**Homepage BEFORE**
homepage_seo-meta_BEFORE

**Homepage AFTER**
homepage_seo-meta_AFTER

<br/>
<br/>
### Providers

**Providers page BEFORE**
providers-page_seo-meta_BEFORE

**Providers page AFTER**

providers-page_seo-meta_AFTER

<br/>
<br/>
### Course pages

**Course page BEFORE**
course-page_seo-meta_BEFORE

**Course page AFTER**
course-page_seo-meta_AFTER

### See all screenshots

[seo-meta_improvements_screenshots.zip](https://github.com/Qiskit/qiskit.org/files/12007273/seo-meta_improvements_screenshots.zip)

### CODE REVIEW

**Review by y4izus — CHANGES_REQUESTED** [review]
Thank you Randy! Although there are no limit for the length of the meta description, I suggest to fit them in less than 170 characters because if the users can read the most part of it, they can understand better what the page is for

File: pages/learn/course/algorithm-design.vue [hunk 1]
```diff
@@ -28,17 +28,32 @@ definePageMeta({
...
-useHead({
-  title: "Variational Algorithm Design",
+const config = useRuntimeConfig();
+
+// SEO specific definitions
+const title = "Variational algorithm design";
+const description =
+  "This course teaches how to write variational algorithms: near-term, hybrid-quantum-classical algorithms that are ideal candidates to achieve quantum advantage.";
+const image = `${config.public.siteUrl}/images/learn/course/algorithm-design/algorithm-design.png`;
+const pageUrl = `${config.public.siteUrl}/learn/course/algorithm-design`;
```
[the remark this answers is not in the document]
Comment (techtolentino) [reply pages/learn/course/algorithm-design.vue:38]: this happened in a previous update for QGSS23 and Twitter cards - without the `https://qiskit.org`, the images don't show up

File: pages/learn/summer-school/quantum-simulation-summer-school-2022.vue [hunk 2]
```diff
@@ -27,21 +27,37 @@ definePageMeta({
...
 });
 
-useHead({
-  title: "2022 Qiskit global summer school on quantum simulation",
+const config = useRuntimeConfig();
+
+// SEO specific definitions
+const title = "2022 Qiskit global summer school on quantum simulation";
+const description =
+  "This summer school provides a focused introduction to quantum computing and its applications to quantum simulation, with a specific focus on quantum chemistry. These lectures were first released in July 2022.";
```
[the remark this answers is not in the document]
Comment (techtolentino) [reply pages/learn/summer-school/quantum-simulation-summer-school-2022.vue]: Resolved in https://github.com/Qiskit/qiskit.org/pull/3407/commits/c0a07df7984d15c1f296194cb0ae22a5bd2ce3ef

File: pages/learn/course/quantum-computing-labs.vue [hunk 3]
```diff
@@ -39,17 +39,33 @@ definePageMeta({
...
 });
 
-useHead({
-  title: "Quantum Computing Labs",
+const config = useRuntimeConfig();
+
+// SEO specific definitions
+const title = "Quantum Computing Labs";
+const description =
+  "These are hands-on exercises that interact with IBM Quantum’s real quantum systems. Lab modules cover basic quantum circuits, to exploring core algorithms such as Grover search and Shor's algorithm.";
```
[the remark this answers is not in the document]
Comment (techtolentino) [reply pages/learn/course/quantum-computing-labs.vue]: Resolved in https://github.com/Qiskit/qiskit.org/pull/3407/commits/13e96493de301d8fe5513871a5306f5c0b9ef6f6

File: pages/learn/course/machine-learning-course.vue [hunk 4]
```diff
@@ -27,13 +27,30 @@ definePageMeta({
...
 });
 
-useHead({
-  title: "Quantum machine learning course",
+const config = useRuntimeConfig();
+
+// SEO specific definitions
+const title = "Quantum machine learning course";
+const description =
+  "This eight-hour will take you through key concepts in quantum machine learning, such as parameterized quantum circuits, training circuits, and applying them to basic problems.";
```
[the remark this answers is not in the document]
Comment (techtolentino) [reply pages/learn/course/machine-learning-course.vue]: Resolved in https://github.com/Qiskit/qiskit.org/pull/3407/commits/13e96493de301d8fe5513871a5306f5c0b9ef6f6

File: pages/learn/course/legacy-content.vue [hunk 5]
```diff
@@ -39,13 +39,30 @@ definePageMeta({
...
 });
 
-useHead({
-  title: "Legacy content",
+const config = useRuntimeConfig();
+
+// SEO specific definitions
+const title = "Legacy content";
+const description =
+  "Access older versions of the Qiskit Textbook through our versioning system. Stable pages of the original version (v1) are available. Note: They may be outdated and not compatible with current Qiskit versions.";
```
[the remark this answers is not in the document]
Comment (techtolentino) [reply pages/learn/course/legacy-content.vue]: Resolved in https://github.com/Qiskit/qiskit.org/pull/3407/commits/13e96493de301d8fe5513871a5306f5c0b9ef6f6

File: pages/learn/course/introduction-course.vue [hunk 6]
```diff
@@ -28,13 +28,29 @@ definePageMeta({
...
 });
 
-useHead({
-  title: "Introduction course",
+const config = useRuntimeConfig();
+
+// SEO specific definitions
+const title = "Introduction course";
+const description =
+  "This short 3-hour course contains is aimed at self-learners from all backgrounds, and gives a solid understanding of the principles behind quantum computing, focusing on developing intuition.";
```
[the remark this answers is not in the document]
Comment (techtolentino) [reply pages/learn/course/introduction-course.vue]: Resolved in https://github.com/Qiskit/qiskit.org/pull/3407/commits/13e96493de301d8fe5513871a5306f5c0b9ef6f6

File: pages/learn/course/basics-quantum-information.vue [hunk 7]
```diff
@@ -28,21 +28,37 @@ definePageMeta({
...
 });
 
-useHead({
-  title: "Basics of quantum information",
+const config = useRuntimeConfig();
+
+// SEO specific definitions
+const title = "Basics of quantum information";
+const description =
+  "This free course covers quantum information at a detailed mathematical level. Join John Watrous as you explore quantum information, quantum algorithms, and how to understand and mitigate noise.";
```
[the remark this answers is not in the document]
Comment (techtolentino) [reply pages/learn/course/basics-quantum-information.vue]: Resolved in https://github.com/Qiskit/qiskit.org/pull/3407/commits/13e96493de301d8fe5513871a5306f5c0b9ef6f6

File: pages/events/summer-school-2022.vue [hunk 8]
```diff
@@ -115,61 +115,25 @@ definePageMeta({
   routeName: "summer-school-2022",
 });
 
+const config = useRuntimeConfig();
```
[the remark this answers is not in the document]
Comment (techtolentino) [reply pages/events/summer-school-2022.vue:118]: Resolved in https://github.com/Qiskit/qiskit.org/pull/3407/commits/13e96493de301d8fe5513871a5306f5c0b9ef6f6

File: pages/events/index.vue [hunk 9]
```diff
@@ -135,16 +135,19 @@ definePageMeta({
 const pastEvents = rawPastEvents as CommunityEvent[];
 const upcomingEvents = rawUpcomingEvents as CommunityEvent[];
 const config = useRuntimeConfig();
+const description =
```
[the remark this answers is not in the document]
Comment (techtolentino) [reply pages/events/index.vue:138]: Resolved in https://github.com/Qiskit/qiskit.org/pull/3407/commits/13e96493de301d8fe5513871a5306f5c0b9ef6f6

**Pull request merged**